vcpkg/src/lib.rs

```diff
@@ -398,19 +398,196 @@ fn find_vcpkg_target(cfg: &Config, target_triplet: &TargetTriplet) -> Result<Vcp
-        is_static: target_triplet.is_static,
         status_path: status_path,
-        lib_suffix: target_triplet.lib_suffix.to_owned(),
-        strip_lib_prefix: target_triplet.strip_lib_prefix,
+        packages_path: packages_path,
+        target_triplet: target_triplet.clone(),
     })
 }
 
+/// Parsed knowledge from a .pc file.
+#[derive(Debug)]
+struct PcFile {
+    /// The pkg-config name of this library.
+    id: String,
+    /// List of libraries found as '-l', translated to a given vcpkg_target. e.g. libbrotlicommon.a
+    libs: Vec<String>,
+    /// List of pkgconfig dependencies, e.g. PcFile::id.
+    deps: Vec<String>,
+}
+impl PcFile {
+    fn parse_pc_file(vcpkg_target: &VcpkgTarget, path: &Path) -> Result<PcFile, Error> {
+        // Extract the pkg-config name.
+        let id = try!(path
+            .file_stem()
+            .ok_or_else(|| Error::VcpkgInstallation(format!(
+                "pkg-config file {} has bogus name",
+                path.to_string_lossy()
+            ))))
+        .to_string_lossy();
+        // Read through the file and gather what we want.
+        let mut file = try!(File::open(path)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't open {}", path.display()))));
+        let mut pc_file_contents = String::new();
+
+        try!(file
+            .read_to_string(&mut pc_file_contents)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't read {}", path.display()))));
+        PcFile::from_str(&id, &pc_file_contents, &vcpkg_target.target_triplet)
+    }
+    fn from_str(id: &str, s: &str, target_triplet: &TargetTriplet) -> Result<Self, Error> {
+        let mut libs = Vec::new();
+        let mut deps = Vec::new();
+
+        for line in s.lines() {
+            // We could collect alot of stuff here, but we only care about Requires and Libs for the moment.
+            if line.starts_with("Requires:") {
+                let requires_args = line.split(":").skip(1).next().unwrap_or("").split(",");
+                for mut dep in requires_args {
+                    // Drop any versioning requirements, we only care about library order and rely upon
+                    // port dependencies to resolve versioning.
+                    if let Some(inequality) = dep.find(|c| c == '=' || c == '<' || c == '>') {
+                        dep = dep.split_at(inequality).0;
+                    }
+                    dep = dep.trim();
+                    // Sometimes Requires: lines don't have commas, so split again.
+                    deps.extend(dep.split_whitespace().map(|s| s.to_owned()));
+                }
+            } else if line.starts_with("Libs:") {
+                let lib_flags = line
+                    .split(":")
+                    .skip(1)
+                    .next()
+                    .unwrap_or("")
+                    .split_whitespace();
+                for lib_flag in lib_flags {
+                    if lib_flag.starts_with("-l") {
```

Comment:
I think Clippy would tell you to use `lib_flag.strip_prefix()` here, which lets you get the prefix test and the de-prefixed string in one shebang.

Reply:
Yeah I believe I started with strip_prefix() but changed to open-form start_with after realizing that it only came about on 1.45, breaking down version rustc compatibility.